catroid/src/main/java/org/catrobat/catroid/ui/SettingsActivity.java

```diff
@@ -256,6 +258,23 @@ private void setNXTSensors() {
 		}
 	}
 
+	@SuppressWarnings("deprecation")
+	private void setHintPreferences() {
+		CheckBoxPreference hintCheckBoxPreference = (CheckBoxPreference) findPreference(SETTINGS_SHOW_HINTS);
+
+		if (!BuildConfig.SNACKBAR_HINTS_ENABLED) {
```

Comment:
`hintCheckBoxPreference.setChecked(BuildConfig.SNACKBAR_HINTS_ENABLED);` should work here too